arbitrator/prover/src/value.rs

```diff
@@ -86,6 +86,33 @@ impl ProgramCounter {
     }
 }
 
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SmallProgramCounter {
```

Comment:
Instead of creating a `SmallProgramCounter`, what if we changed the `ProgramCounter` to use `u32`, ensured consistency with serialization, and then added some methods like `module()` that return a `usize` for indexing